include/irr/asset/IAsset.h

```diff
@@ -21,9 +21,51 @@ class IAssetMetadata : public core::IReferenceCounted
     virtual const char* getLoaderName() = 0;
 };
 
+//! An interface class representing any kind of cpu-objects with capability of being cached
+/**
+	Actually it can be anything like cpu-side meshes scenes, texture data and material pipelines, 
+	but must be serializable into/from .baw file format, unless it comes from an extension (irr::ext), 
+	so forth cached. There are different asset types you can find at IAsset::E_TYPE. 
+	IAsset doesn't provide direct instantiation (virtual destructor), much like IReferenceCounted.
+
+	Every asset must be loaded by a particular class derived from IAssetLoader.
```

Comment:
Not true, IAsset is just an ADT, so its not an object in itself (cannot be instantiated). But concrete classes (such as ICPUBuffer, ICPUMeshBuffer, etc.) derived from `IAsset` can be instantiated via `new` or `core::make_smart_refctd_ptr`

Reply:
still need a word about that you can create them outside of loaders

Reply:
gonna cover it soon

Reply:
status?

Reply:
Solved